### PR TITLE
Get ad stats

### PR DESCRIPTION
**Pull request by kesla**
Support http://api.adform.com/Services/Documentation/CampaignStatsService.htm#op.IDAGPQF

### CODE REVIEW

**Comment by nchrisdk** [comment]
this is a style thing... when we deal with namespaces and prefixes, I think it would make the code cleaner to extract all the namespaces and prefixes into separate file. This way we are consistent with the prefixes as well...

File: adform.js [hunk 1]
```diff
@@ -30,53 +32,126 @@ adform.createAd = function(ticket, campaign, name, data, callback) {
...
-                'ns:IgnoreWarnings': true 
+                'ns:IgnoreWarnings': true
             }
         }
     }, callback);
-}
+};
 
 adform.getCampaigns = function(ticket, advertiser, callback) {
+    function formatCampaigns(rawCampaigns) {
```
Comment (nchrisdk) [inline adform.js]: function not used

File: adform.js [hunk 2]
```diff
@@ -30,53 +32,126 @@ adform.createAd = function(ticket, campaign, name, data, callback) {
...
         data: {
             'ns:GetCampaignsData': {
                 'ns:AdvertiserId': advertiser.id
             }
-        }   
+        }
     }, function(err, etree) {
-        var campaigns = etree.findall('./s:Body/Campaigns/Campaign').map(function(campaign) {
+        var rawCampaigns = etree.findall('./s:Body/Campaigns/Campaign');
+        var campaigns = rawCampaigns.map(function(campaign) {
```
Comment (nchrisdk) [inline adform.js:58]: I suspect that the partial function created above was meant to replace this callback...

File: adform.js [hunk 3]
```diff
@@ -30,53 +32,126 @@ adform.createAd = function(ticket, campaign, name, data, callback) {
...
                 endDate: new Date(campaign.findtext('EndDate'))
-            }
+            };
         });
         callback(null, campaigns);
     });
-}
+};
+
+adform.getAdStats = function(ticket, campaign, startDate, endDate, callback) {
```
Comment (nchrisdk) [inline adform.js:70]: a comment or two about how the startDate and endDate values should be formated: 2010-07-23

Comment (kesla) [inline adform.js:70]: nope, startDate and endDate is Date:s. See further down, they are then being transformed to that format.

**Comment by kesla** [comment]
@nchrisdk For the style thing (separating out namespaces) - I agree. But let's do that as a separate task further on. Do you mind create an issue for that?

File: adform.js (hunk 2 above)
Comment (kesla) [inline adform.js:58]: true, but skip that for now...

**Pull request merged**